src/backend/core/api/viewsets.py

```diff
@@ -209,10 +227,98 @@ def retrieve(self, request, *args, **kwargs):
                 },
             }
         else:
+            analytics.track(
+                user=self.request.user,
+                event="Get Room",
+                properties={"slug": instance.slug},
+            )
             data = self.get_serializer(instance).data
 
         return drf_response.Response(data)
 
+    @decorators.action(detail=True, methods=["post"], url_path="start-recording")
+    def start_room_recording(self, request, pk=None):  # pylint: disable=unused-argument
+        """Start room recording."""
+        if not settings.ENABLE_RECORDING:
+            return drf_response.Response(
+                {"error": "Recording is disabled."},
+                status=drf_status.HTTP_403_FORBIDDEN,
+            )
```

Comment:
NIT: You may directly do 
```suggestion
            raise PermissionDenied({"error": "Recording is disabled."})
```